### PR TITLE
[alt] remove default iree-compile extra args from onnx-iree mode

### PR DESCRIPTION
**Pull request by zjgarvey**
Always using `iree-input-demote-i64-to-i32` was causing numerics problems since it wasn't loading i64 inputs correctly. Additionally the flags for specifying the chip for `hip` is out of date, and should be removed. Users should specify `-ica "iree-hip-target=gfx942" "other-arg" "other-arg" ...` if compiling for Mi300.

### CODE REVIEW

File: alt_e2eshark/e2e_testing/backends.py
Comment (ScottTodd) [inline alt_e2eshark/e2e_testing/backends.py:39]: Got more context for this? Something doesn't seem right here...

also typo: "deprecated" not "depreciated"

File: alt_e2eshark/e2e_testing/backends.py
Comment (ScottTodd) [inline alt_e2eshark/e2e_testing/backends.py:98]: Note that this `iree-hal-target-device` flag is quite new, and we aren't even using it in-tree in iree-org/iree for many tests yet.

File: alt_e2eshark/e2e_testing/backends.py
Comment (zjgarvey) [reply alt_e2eshark/e2e_testing/backends.py:39]: I'm looking at CompilerOptions and, for example, <https://github.com/iree-org/iree/blob/35dafeeb73ad7ace5b81b6250c3c29f6809ee2b4/compiler/bindings/python/iree/compiler/tools/core.py#L187>. This still uses `--iree-hal-target-backends` rather than `-iree-hal-target-device`. 

Will change the typo, thanks.

**Comment by zjgarvey** [comment]
@ScottTodd Should I revert the change to the cl config to use `--iree-hal-target-backends` instead of `--iree-hal-target-device`?

**Comment by ScottTodd** [comment]
> @ScottTodd Should I revert the change to the cl config to use `--iree-hal-target-backends` instead of `--iree-hal-target-device`?

That's probably safer for now. Since the new flag hasn't been fully adopted in the IREE repo and doesn't have much mileage on it, it may be buggy. That's going to be easier to reason about upstream first, and that sort of flag change can also be made independently from the rest of the cleanup here.

**Review by vinayakdsci — APPROVED** [review]
LGTM, thanks! But let's wait for an approval from @ScottTodd too.

File: alt_e2eshark/e2e_testing/backends.py
Comment (ScottTodd) [inline alt_e2eshark/e2e_testing/backends.py:140]: FYI, we've been pulling this from an environment variable in other tests. See https://github.com/nod-ai/SHARK-Platform/pull/373#discussion_r1822904806 for some discussion

**Review by ScottTodd — APPROVED** [review]

File: alt_e2eshark/e2e_testing/backends.py
Comment (zjgarvey) [reply alt_e2eshark/e2e_testing/backends.py:140]: Yeah, that's a good idea. It's a bit verbose to need to write out all the flags repeatedly.

**Pull request merged**
